Automattic/simplenote-ios
# Notes List: New Swipeable Actions

### Details:
This PR implements new Swipeable actions over the Notes List: Share / Pin and Unpin!

Closes #339

### Testing:
1. Launch Simplenote
2. Swipe over any of the Notes and....
2.1. Press over the `Share` action: Verify that the Share Sheet shows up
2.2. Press over the `Pin` action: Verify it gets Pinned / Unpinned
2.3. Press over the `Trash` action: Verify the note gets trashed

### Testing:
1. Launch simplenote
2. Trash any note
3. Open the **Trash**
4. Swipe over any of the notes and verify the standard `Restore / Delete` actions show up
